src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -635,6 +647,16 @@ private static Expression CreateArgument(ParameterInfo parameter, FactoryContext
         {
             return RequestPipeReaderExpr;
         }
+        else if (parameter.CustomAttributes.Any(a => typeof(AsParametersAttribute).IsAssignableFrom(a.AttributeType)))
```

Comment:
Can we put this above the `ParameterType` checks with all the other `parameterCustomAttributes`? I know it might be weird to look at the attribute before the well-known type. I can't think of any well-known type that it makes sense to interpret differently than how we treat it by default, but we're consistent with attributes always overriding all other inference and I want to be consistent about that.

We already evaluated `CustomAttributes` into the `parameterCustomAttributes` variable, so we should reuse that. Also, since `AsParametersAttribute` is sealed, can we just do a normal equality comparison? So after moving this earlier, it would be.

```suggestion
        else if (parameterCustomAttributes.Any(a => a.AttributeType == typeof(AsParametersAttribute)))
```